src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -0,0 +1,353 @@
+				getDocumentService().routeDocument(dv, "Disbursement Document created from recurring DV: " + recurringDV.getDocumentHeader().getDocumentNumber(), adHocRoutingRecipients);
+				getDocumentService().blanketApproveDocument(dv, "Disbursement Document automatically approved by recurring DV", adHocRoutingRecipients);
+				getBusinessObjectService().save(recurringDV.getRecurringDisbursementVoucherDetails());
+			} catch (WorkflowException e) {
+				LOG.error("There was an errorr trying to save our route the created Disbursement Voucher documents: ", e);
+				throw new RuntimeException(e);
+			}
+		}
+	}
+	
+	@Override
+	public List<RecurrningDisbursementVoucherPDPStatus> getPdpStatuses(RecurringDisbursementVoucherDocument recurringDV) {
+	    List<RecurrningDisbursementVoucherPDPStatus> pdpStatuses = new ArrayList<RecurrningDisbursementVoucherPDPStatus>();
+	    for (RecurringDisbursementVoucherDetail detail : recurringDV.getRecurringDisbursementVoucherDetails()) {
+	        if (!StringUtils.isEmpty(detail.getDvDocumentNumber())){ 
+	            DisbursementVoucherDocument dv;
+                try {
+                    dv = (DisbursementVoucherDocument) getDocumentService().getByDocumentHeaderId(detail.getDvDocumentNumber());
+                } catch (WorkflowException e) {
+                    LOG.error("getPdpStatuses: There was a problem getting DV from the recurring DV detail: " + e);
+                    throw new RuntimeException(e);
+                }
+	            RecurrningDisbursementVoucherPDPStatus pdpStatus = new RecurrningDisbursementVoucherPDPStatus();
+	            pdpStatus.setDocumentNumber(dv.getDocumentNumber());
+	            pdpStatus.setCancelDate(dv.getCancelDate());
+	            pdpStatus.setExtractDate(dv.getExtractDate());
+	            pdpStatus.setPaidDate(dv.getPaidDate());
+	            pdpStatus.setPdpStatus(dv.getDisbursementVoucherPdpStatus());
+	            pdpStatus.setPaymentDetailDocumentType(dv.getPaymentDetailDocumentType());
+	            pdpStatus.setDueDate(dv.getDisbursementVoucherDueDate());
+	            pdpStatuses.add(pdpStatus);
+	        }
+	    }
+	    Collections.sort(pdpStatuses);
+	    return pdpStatuses;
+	}
+	
+	@Override
+	public Set<String> cancelPDPPayments(RecurringDisbursementVoucherDocument recurringDisbursementVoucherDocument, String cancelMessage) {
+	    Collection<PaymentDetail> paymentDetails = getPaymentDetailsFromRecurringDisbursementVoucher(recurringDisbursementVoucherDocument);
+	    Set<String> canceledPaymentGroups = new HashSet<String>();
+	    for (PaymentDetail detail : paymentDetails) {
+	        String paymentDetailStatus = detail.getPaymentGroup().getPaymentStatusCode();
+	        boolean doCancel = paymentDetailStatus != null  && isPaymentCancelable(GlobalVariables.getUserSession().getPerson(), 
+	                paymentDetailStatus);
+	        if (doCancel) {
+	            LOG.info("About to Cancel " + detail.getId() + " for a reason of " + cancelMessage);
```

Comment:
Are these two log messages about canceling appropriate at INFO level, or should they be DEBUG?